src/client/method-list-units.c

```diff
@@ -153,18 +157,79 @@ void unit_list_unref(UnitList *unit_list) {
         free(unit_list);
 }
 
-/* Prints 100-characters-wide table */
+void calc_max_col_len(const char *data, unsigned long *max_len) {
+        unsigned long len = strlen(data);
+        if (len > *max_len) {
+                *max_len = len;
+        }
+}
+
+unsigned int calc_line_len(const unsigned int sep_len, const unsigned int num_of_cols, ...) {
+        unsigned int len = 0;
+        va_list args;
+
+        va_start(args, num_of_cols);
+        for (unsigned int i = 0; i < num_of_cols; i++) {
+                len += va_arg(args, unsigned long); // max column length
+                len += sep_len;                     // length of column separator string
+        }
+        va_end(args);
+
+        if (len > 0) {
+                // at least one column exists, remove column separator at the end
+                len -= sep_len;
+        }
+        return len;
+}
```

Comment:
This function is only used once and much of its complexity stems from being a variadic function. Therefore, I'd get rid of this function and do this in-place in `print_unit_list_simple`.